specs/TrackingPrevention.md

```diff
@@ -0,0 +1,205 @@
+  /// You can set this property to false to disable the tracking prevention feature if the app only renders
+  /// content in the WebView2 that is known to be safe. Disabling this feature when creating environment also
+  /// improves runtime performance by skipping related code.
+  ///
+  /// You shouldn't disable this property if WebView2 is being used as a "full browser" with arbitrary navigation
+  /// and should protect end user privacy.
+  ///
+  ///
+  /// There is `TrackingPreventionLevel` property to control levels of tracking prevention of the WebView2's
+  /// associated with a same user profile. However, you can also turn off tracking prevention later using 
+  /// `TrackingPreventionLevel` property and `COREWEBVIEW2_TRACKING_PREVENTION_LEVEL_OFF` value
+  /// but that doesn't improves runtime performance.
+  /// See `TrackingPreventionLevel` for more details.
+  ///
+  /// Tracking prevention protects users from online tracking by restricting
+  /// the ability of trackers to access browser-based storage as well as the network.
+  /// See [Tracking prevention](microsoft-edge/web-platform/tracking-prevention).
+  [propget] HRESULT IsTrackingPreventionEnabled([out, retval]  BOOL* value);
+  /// Sets the `IsTrackingPreventionEnabled` property.
+  [propput] HRESULT IsTrackingPreventionEnabled([in]  BOOL value);
+}
+
+/// This is the ICoreWebView2 profile.
+[uuid(ddc4070a-c873-11eb-b8bc-0242ac130003), object, pointer_default(unique)]
+interface ICoreWebView2Profile5: IUnknown {
+  /// The `TrackingPreventionLevel` property sets the level of tracking prevention when `IsTrackingPreventionEnabled`
+  /// is enabled and otherwise fails with `E_ABORT`. This level would apply to the context of the user profile.
+  /// That is, other WebViews under the same user profile could be affected.
+  ///
+  /// See `COREWEBVIEW2_TRACKING_PREVENTION_LEVEL` for descriptions of levels.
+  ///
+  /// If tracking prevention feature is enabled when creating the WebView2 environment, you can also turn off
+  /// tracking prevention later using `TrackingPreventionLevel` property and `COREWEBVIEW2_TRACKING_PREVENTION_LEVEL_OFF` value
+  /// but that doesn't improves runtime performance.
+  ///
+  /// There is `IsTrackingPreventionEnabled` property to toggle tracking prevention feature for all the WebView2's
+  /// created in the same environment. If enabled, `TrackingPreventionLevel` is set to `COREWEBVIEW2_TRACKING_PREVENTION_LEVEL_BALANCED` by default
+  /// for all the WebView2's and profiles created in the same enviroment. If disabled `TrackingPreventionLevel`
+  /// will be `COREWEBVIEW2_TRACKING_PREVENTION_LEVEL_OFF` and cannot be changed to other levels.
+  /// See `IsTrackingPreventionEnabled` for more details.
+  [propget] HRESULT TrackingPreventionLevel(
+      [out, retval] COREWEBVIEW2_TRACKING_PREVENTION_LEVEL* value);
+  /// Set the `TrackingPreventionLevel` property.
+  ///
+  /// Returns with `E_ABORT` if `IsTrackingPreventionEnabled` is false.
```

Comment:
@david-risney If the feature is turned off from environment and they are trying to set the level, should we allow ignore it silently in our code or throw failure with `E_ABORT`?

Reply:
Good question. I'd go with allow the property to be modified but changes do nothing if IsTrackingPreventionEnabled is false. Either way, good to call it out here, and we can ask API board folks what they think.